samples/bluetooth/peripheral_identity/src/peripheral_identity.c

```diff
@@ -61,6 +64,21 @@ static void adv_start(struct k_work *work)
 	printk("Using current id: %u\n", id_current);
 	adv_param.id = id_current;
 
+#if defined(CONFIG_BT_OBSERVER) && defined(CONFIG_BT_PRIVACY)
```

Comment:
I think it's better to not if-def this out. The test becomes easier to follow if it always does the same thing.

Reply:
Reverted without `CONFIG_BT_PRIVACY`:
```
/opt/toolchains/zephyr-sdk-0.17.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(peripheral_identity.c.obj): in function `adv_start':
/__w/zephyr/zephyr/samples/bluetooth/peripheral_identity/src/peripheral_identity.c:69: undefined reference to `bt_le_scan_stop'
```